olive-writing.md

```diff
@@ -0,0 +1,124 @@
+# How to create actions for Shesmu to launch
+
+1. [Configure Shesmu to find actions and olives](#configure-shesmu)
+1. [Write olives to open or close JIRA tickets](#write-olives-for-jira-ticket-actions)
+1. [Write custom actions and olives](#write-olives-for-custom-actions)
+
+
+## Configure Shesmu
+All of these files should be placed in the Shesmu server directory (same directory as the Shesmu JAR).
+### JIRA config files
+Each project in your JIRA must be configured using a separate file named `<project-name>.jira` with the following format:
+```
+{
+  "projectKey": <code for project>,
+  "token": <API token for project>,
+  "url": <URL of the JIRA instance>
+}
+```
+If too many JIRA actions are launched at once, JIRA may (blacklist? throttle?) more actions from that API key. This can be avoided by adding in a `jira.ratelimit` file to the Shesmu server folder, which tells Shesmu to limit the number of JIRA actions it launches. The following `jira.ratelimit` example file lists some useful defaults:
+```
+{
+  "capacity": 50,
+  "delay": 500
+}
+```
+
+### Action files
```

Comment:
This only applies if you have all the Guanyin infrastructure plus a bunch of stuff that we haven't released publically...